graviti/schema/catalog_to_schema.py

```diff
@@ -0,0 +1,322 @@
+    "array": "array",
+    "boolean": "boolean",
+    "integer": "int32",
+    "number": "float32",
+    "string": "string",
+}
+
+
+_PYTHON_TYPE_TO_PORTEX_TYPE = {
+    int: "int32",
+    float: "float32",
+}
+
+_LABEL_PROCESSORS: Dict[str, "_ConverterBase"] = {}
+
+_SUFFIX_TO_FILE_FIELDS: Dict[str, Dict[str, Any]] = {}
+
+
+def _fill_file_fields() -> None:
+    _SUFFIX_TO_FILE_FIELDS[".bin"] = {"name": "point_cloud", "type": "file.PointCloudBin"}
+    for suffix in (".pcd", ".ply", ".las", ".laz"):
+        _SUFFIX_TO_FILE_FIELDS[suffix] = {"name": "point_cloud", "type": "file.PointCloud"}
+    for suffix in (".bmp", ".jpg", ".jpeg", ".png", ".gif"):
+        _SUFFIX_TO_FILE_FIELDS[suffix] = {"name": "image", "type": "file.Image"}
+    for suffix in (".mp3", ".wav", ".wma", ".aac"):
+        _SUFFIX_TO_FILE_FIELDS[suffix] = {"name": "audio", "type": "file.Audio"}
+
+
+_fill_file_fields()
+
+
+def _get_category_params(subcatalog: Union[CategoriesMixin, MaskCategoriesMixin]) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "categories"):
+        return {}
+
+    category_params: Dict[str, Any] = {}
+    if isinstance(subcatalog, MaskCategoriesMixin):
+        category_params["categoryIds"] = subcatalog.get_index_to_category()
+    else:
+        category_params["categories"] = list(subcatalog.categories.keys())
+
+    category_delimiter = getattr(subcatalog, "category_delimiter", None)
+    if category_delimiter:
+        category_params["categoryDelimiter"] = category_delimiter
+
+    return category_params
+
+
+def _get_attribute_params(subcatalog: AttributesMixin) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "attributes"):
+        return {}
+
+    attribute_params: List[Dict[str, Any]] = []
+    for attribute in subcatalog.attributes:
+        field: Dict[str, Any] = {"name": attribute.name}
+        if hasattr(attribute, "enum"):
+            field["type"] = "enum"
+            field["values"] = attribute.enum
+        else:
+            attribute_type = (
+                reduce(lambda a, b: a or b, attribute.type)
+                if isinstance(attribute.type, list)
+                else attribute.type
+            )
+            field["type"] = _JSON_TYPE_TO_PORTEX_TYPE[attribute_type]
+
+            if hasattr(attribute, "minimum"):
+                field["minimum"] = attribute.minimum
+            if hasattr(attribute, "maximum"):
+                field["maximum"] = attribute.maximum
+        attribute_params.append(field)
+    return {"attributes": attribute_params}
+
+
+def _get_category_and_attribute_params(subcatalog: Any) -> Dict[str, Any]:
+    return {**_get_category_params(subcatalog), **_get_attribute_params(subcatalog)}
+
+
+def _get_dtype(value: float, default_type: str = "int32") -> Dict[str, str]:
+    dtype = _PYTHON_TYPE_TO_PORTEX_TYPE[type(value)]
+    return {"dtype": dtype} if dtype != default_type else {}
+
+
+class _ConverterBase:  # pylint: disable=too-few-public-methods
+    import_types: Any
```

Comment:
why we need this type hint?

Reply:
To declare the attribute `import_types` for the use in `_get_label_fields`.